wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -523,23 +523,29 @@ void recover() {
       LOG.warn("Recovery of the workspaces is rejected.");
       return;
     }
+    Set<RuntimeIdentity> identities;
     try {
-      for (RuntimeIdentity identity : infrastructure.getIdentities()) {
-        recoverOne(infrastructure, identity);
-      }
-    } catch (UnsupportedOperationException x) {
+      identities = infrastructure.getIdentities();
+    } catch (UnsupportedOperationException e) {
       LOG.warn("Not recoverable infrastructure: '{}'", infrastructure.getName());
-    } catch (InternalInfrastructureException x) {
-      LOG.error(
-          format(
-              "An error occurred while attempted to recover runtimes using infrastructure '%s'",
-              infrastructure.getName()),
-          x);
-    } catch (ConflictException | ServerException | InfrastructureException x) {
+      return;
+    } catch (InfrastructureException e) {
       LOG.error(
           "An error occurred while attempted to recover runtimes using infrastructure '{}'. Reason: '{}'",
```

Comment:
Le'ts rephrase this message to make it clear that error occurred during identities fetching.

Reply:
Done!